SwissDataScienceCenter/renku-python
# feat(service): create new projects from templates

Adds the full `renku init` functionality to the service. It always requires a remote repository
containing valid template since there is no locally cached version. Fix #862

Please note that the user provided in the fixture `authentication_headers` must be able to create projects for the tests to succeed.

**How to test**
A temporary preview is deployed here: https://lorenzo.dev.renku.ch (service API reachable at `/api/renku/`)

You can use Postman to invoke the APIs. Always include the following headers
```
x-requested-with: XMLHttpRequest
Content-Type: application/json
Cookie: session=<your-valid-session-cookie-here>
```

READ THE MANIFEST
You can check the available templates.
```
GET https://lorenzo.dev.renku.ch/api/renku/templates.read_manifest?url=https://github.com/SwissDataScienceCenter/renku-project-template
```

CREATE PROJECT
Create a project by providing the proper data in the body.
```
POST https://lorenzo.dev.renku.ch/api/renku/templates.create_project
```

Assume you want to create a new project from the python template, here is a stub of the body (adjust `namespace` and `name`)
```
{
	"url": "https://github.com/SwissDataScienceCenter/renku-project-template",
	"ref": "0.1.10",
	"identifier": "python-minimal",
	"parameters": [{
		"key": "description",
		"value": "put your description here"
	}],
	"project_name": "test from postman",
	"project_namespace": "your namespace here",
	"project_repository": "https://dev.renku.ch/gitlab"
}
```
